src/processors/universal.ts

```diff
@@ -737,6 +747,10 @@ while calculateExpectedTakeAmount returned ${tokenAddressToString(orderInfo.orde
       return;
     }
 
+    if (orderIsConfirmed) {
+      this.unconfirmedOrdersBudgetController.validateAndAddOrder(orderId, usdWorth, logger);
+    }
+
```

Comment:
This is nonsense: you literally add confirmed order (isConfirmed==true) to the list of unconfirmed orders. How can this happen? 

Also, why do you perform validation after the order has been fulfilled, why?